### ISSUE TITLE
Failure seen in 3ru_3df automated integration test with recent changes

### ISSUE THREAD

**Issue by bieryAtFnal** [DUNE-DAQ/daqconf#273]
It seems that the most recent changes in the `daqconf` repo have broken the `dfmodules/integtest/3ru_3df_multirun_test.py` test.

The most recent changes in `daqconf` seem to be related to a TPZipper cardinality bug fix, so I will assign this to @CharlieBatchelor , but it would be great if @plasorak and @eflumerf could provide assistance, if needed.

Here are instructions that produce a successful running of the 3ru_3df test:
```
source /cvmfs/dunedaq.opensciencegrid.org/setup_dunedaq.sh
setup_dbt latest
dbt-create -n N23-01-18 18JanDevTest
cd 18JanDevTest
cd sourcecode
git clone https://github.com/DUNE-DAQ/daqconf.git -b develop
cd daqconf
git checkout 72c8947
cd ..
git clone https://github.com/DUNE-DAQ/daq-systemtest.git -b develop
git clone https://github.com/DUNE-DAQ/dfmodules.git -b develop
cd ..
dbt-workarea-env
git clone https://github.com/DUNE-DAQ/nanorc.git -b develop
cd nanorc ; pip install -U . ; cd ..
dbt-build -j 20
dbt-workarea-env
cd sourcecode/dfmodules/integtest
cp -v `assets-list -c 9f14e12a0ebdaf207e9e740044b2433c | awk '{print $NF}'` .
pytest -s 3ru_3df_multirun_test.py --frame-file $PWD/frames.bin --nanorc-option partition-number 2
```

Here are instructions that produce a failed running of the 3ru_3df test:
```
source /cvmfs/dunedaq.opensciencegrid.org/setup_dunedaq.sh
setup_dbt latest
dbt-create -n N23-01-18 18JanDevTest
cd 18JanDevTest
cd sourcecode
git clone https://github.com/DUNE-DAQ/daqconf.git -b develop
git clone https://github.com/DUNE-DAQ/daq-systemtest.git -b develop
git clone https://github.com/DUNE-DAQ/dfmodules.git -b develop
cd ..
dbt-workarea-env
git clone https://github.com/DUNE-DAQ/nanorc.git -b develop
cd nanorc ; pip install -U . ; cd ..
dbt-build -j 20
dbt-workarea-env
cd sourcecode/dfmodules/integtest
cp -v `assets-list -c 9f14e12a0ebdaf207e9e740044b2433c | awk '{print $NF}'` .
pytest -s 3ru_3df_multirun_test.py --frame-file $PWD/frames.bin --nanorc-option partition-number 2
```

**Comment by bieryAtFnal** [comment]
I should have mentioned that the 3ru_3df test has three parts.  First, it uses a basic system, then it adds SWTPG, then it adds DQM.  The first part works fine in all cases, but the SWTPG and DQM parts fall over fairly hard with the latest daqconf changes.

**Comment by CharlieBatchelor** [comment]
I'm able to replicate the issue. The errors seen in the second round of tests, from where SW TPG is introduced are reminiscent of those seen in the VDCB runs, when the `cardinality` parameter was being incorrectly configured.

If I revert the one line change that was made to `daqconf` [here](https://github.com/DUNE-DAQ/daqconf/pull/270/files#), then the trigger errors are removed from the integration tests above. However, there are many errors of the following type at the end of the last test:

`3ru_3df_multirun_test.py:120: AssertionError`

@bieryAtFnal - Are these normally seen? If not, I suspect there are some other changes that still need to be reverted before we get the expected behaviour back. I'll investigate the cause of the trigger stopping issue in the meantime.

**Comment by CharlieBatchelor** [comment]
I've created a branch which reverts the recent one line change to `trigger_gen.py`, called `cbatchelor/revert_cardinality_changes`. Using this branch of `daqconf` restores the same behaviour I see with the first set of (successful) instructions above. However, these successful instructions are generating 3 `AssertionError` blocks at the end of the my integration tests which @ArturSztuc is able to replicate.

**Comment by CharlieBatchelor** [comment]
It seems the original fix failed to account for the fact that the number of input queues to the `TPZippers` is divided up among them by the number of readout apps. This meant the integration tests with multiple df and ru apps were failing since the `cardinality` parameter was _still_ being set incorrectly. This is fixed in #275 and has been tested against the integration tests Kurt points us to above.

**Comment by bieryAtFnal** [comment]
Closing this Issue as the problem is now fixed, as Charlie describes above.
